src/client/sandbox/storages/wrapper.js

```diff
@@ -0,0 +1,209 @@
+        });
+
+        // NOTE: Save wrapper properties and methods to be able to distinguish them from
+        // properties that will be created from the outside.
+        this.initialProperties = this._getInitialProperties();
+        this.wrapperMethods    = StorageWrapper._getWrapperMethods();
+
+        this._init();
+    }
+
+    static _getWrapperMethods () {
+        var methods = [];
+
+        for (var key in StorageWrapper.prototype)
+            methods.push(key);
+
+        return methods;
+    }
+
+    _init () {
+        this._loadFromNativeStorage();
+        this.lastState = this._getCurrentState();
+
+        window.setInterval(() => this._checkStorageChanged(), 10);
+    }
+
+    _getInitialProperties () {
+        var properties = ['length'];
+
+        for (var property in this) {
+            if (this.hasOwnProperty(property))
+                properties.push(property);
+        }
+
+        return properties;
+    }
+
+    _checkStorageChanged () {
+        var currentState = this._getCurrentState();
+
+        for (var i = 0; i < this.lastState[KEY].length; i++) {
+            var lastStateKey   = this.lastState[KEY][i];
+            var lastStateValue = this.lastState[VALUE][i];
+
+            var keyIndex = currentState[KEY].indexOf(lastStateKey);
+
+            if (keyIndex !== -1) {
+                if (currentState[VALUE][keyIndex] !== lastStateValue)
+                    this._raiseStorageChanged(currentState[KEY][keyIndex], lastStateValue, currentState[VALUE][keyIndex]);
+
+                currentState[KEY].splice(keyIndex, 1);
+                currentState[VALUE].splice(keyIndex, 1);
+            }
+            else
+                this._raiseStorageChanged(lastStateKey, lastStateValue, null);
+        }
+
+        for (var j = 0; j < currentState[KEY].length; j++)
+            this._raiseStorageChanged(currentState[KEY][j], this.EMPTY_OLD_VALUE_ARG, currentState[VALUE][j]);
+
+        this.lastState = this._getCurrentState();
+    }
+
+    _getCurrentState () {
+        var addedProperties = this._getAddedProperties();
+        var state           = [[], []];
+
+        for (var i = 0; i < addedProperties.length; i++) {
+            state[KEY].push(addedProperties[i]);
+            state[VALUE].push(this[addedProperties[i]]);
+        }
+
+        return state;
+    }
+
+    _getAddedProperties () {
+        var properties = [];
+
+        for (var property in this) {
+            if (this.hasOwnProperty(property) && this.initialProperties.indexOf(property) === -1)
+                properties.push(property);
+        }
+
+        return properties;
+    }
+
+    _castToString (value) {
+        // NOTE: The browser automatically translates the key and the value to a string. To repeat this behavior,
+        // we use native storage:
+        // localStorage.setItem(null, null) equivalently to localStorage.setItem('null', 'null')
+        this.nativeStorage[STORAGES_SANDBOX_TEMP] = value;
```

Comment:
Maybe rename to `STORAGE_VALUE_RESOLVER`?.
Is is semantic analog of `DOCUMENT_URL_RESOLVER`